4.56.0 changelog (#4331)
AzureAD/microsoft-authentication-library-for-dotnet: CHANGELOG.md

4.56.0
==========
### New Features

> - MSAL.NET cache extensions (Microsoft.Identity.Client.Extensions.Msal) package has been moved to the main MSAL.NET repository (where any [new issues](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues) should be created). The package version has been increased to match the main MSAL version. Along with this move, support for .NET 4.5 and .NET Core 3.1 was removed and this package now supports only .NET Standard 2.0. Additionally Microsoft.Identity.Client.Extensions.Adal has been deprecated. See [3152](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3152), [4330](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4330).  

Reviewer: ```suggestion
- MSAL.NET cache extensions ([Microsoft.Identity.Client.Extensions.Msal](https://www.nuget.org/packages/Microsoft.Identity.Client.Extensions.Msal)) package has been moved to the main MSAL.NET repository (where any [new issues](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues) should be created). The package version has been increased to match the main MSAL version. Along with this move, support for .NET 4.5 and .NET Core 3.1 was removed and this package now only supports .NET Standard 2.0. Additionally, [Microsoft.Identity.Client.Extensions.Adal](https://www.nuget.org/packages/Microsoft.Identity.Client.Extensions.Adal) has been deprecated. See [3152](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3152), [4330](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4330).
```

> - Added `AuthenticationResult.AuthenticationResultMetadata.Telemetry` that currently contains telemetry from Windows broker (WAM). See [4159](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4159).  

Reviewer: ```suggestion
- Added `AuthenticationResult.AuthenticationResultMetadata.Telemetry` that currently contains telemetry from the [Windows broker (WAM)](https://learn.microsoft.com/entra/msal/dotnet/acquiring-tokens/desktop-mobile/wam). See [4159](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4159).  
```

### Bug Fixes

> - Added throttling logic for acquiring tokens with managed identity to prevent the throttling exceptions thrown by the managed identity endpoints. See [4196](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4196).  

Reviewer 2: this PR will affect both MI and Token Provider using CCA

- Enabled cache synchronization by default. This helps to keep the cache consistent when a singleton confidential client application (CCA) is used with enabled external token cache serialization. The cache synchronization has a negligible performance effect when CCA is created per request. See [4268](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4268).  

> - Fixed an authority validation error in interactive flows when an ADFS authority with a tenant ID was used. See [4272](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4272).  

Reviewer: ```suggestion
- Fixed an authority validation error in interactive flows when an Active Directory Federation Services (ADFS) authority with a tenant ID was used. See [4272](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4272).  
```

- Added clarity to the Windows broker logs. See [4318](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4318).  

4.55.0
==========
### New Features

> - A user assigned managed identity can now be specified using its object ID. See [4215](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4215).  

Reviewer: ```suggestion
- A user-assigned managed identity can now be specified using its object ID. See [4215](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4215).  
```

### Bug Fixes

> - `WithTenantId` now works with CIAM authorities. See [4191](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4191).  

Reviewer: ```suggestion
- [`WithTenantId`](https://learn.microsoft.com/dotnet/api/microsoft.identity.client.abstractapplicationbuilder-1.withtenantid?view=msal-dotnet-latest) now works with CIAM authorities. See [4191](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4191).
```

- MSAL.NET cache extensions ([Microsoft.Identity.Client.Extensions.Msal](https://www.nuget.org/packages/Microsoft.Identity.Client.Extensions.Msal)) package has been moved to the main MSAL.NET repository (where any [new issues](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues) should be created). The package version has been increased to match the main MSAL version. Along with this move, support for .NET 4.5 and .NET Core 3.1 was removed and this package now only supports .NET Standard 2.0. Additionally, [Microsoft.Identity.Client.Extensions.Adal](https://www.nuget.org/packages/Microsoft.Identity.Client.Extensions.Adal) has been deprecated. See [3152](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3152), [4330](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4330).
- Added `AuthenticationResult.AuthenticationResultMetadata.Telemetry` that currently contains telemetry from the [Windows broker (WAM)](https://learn.microsoft.com/entra/msal/dotnet/acquiring-tokens/desktop-mobile/wam). See [4159](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4159).  

### Bug Fixes

> - Added throttling logic for acquiring tokens with managed identity to prevent the throttling exceptions thrown by the managed identity endpoints. See [4196](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4196).  

Author: ```suggestion
- Added throttling logic for acquiring tokens for managed identity (using `AcquireTokenForManagedIdentity` and `WithAppTokenProvider`) to prevent the throttling exceptions thrown by the managed identity endpoints. See [4196](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4196).  
```
